mmengine/infer/infer.py

```diff
@@ -0,0 +1,468 @@
+            metainfo = load(meta_path)
+            for model_cfg in metainfo['Models']:
+                if model_cfg['Name'] == model or model in model_cfg.get(
+                        'Alias', []):
+                    cfg = Config.fromfile(
+                        osp.join(package_path, '.mim', model_cfg['Config']))
+                    weights = model_cfg['Weights']
+                    return cfg, weights
+        raise ValueError(f'Cannot find model: {model} in {project}')
+
+    def _init_model(
+        self,
+        cfg: ConfigType,
+        weights: str,
+        device: str = 'cpu',
+    ) -> nn.Module:
+        """Initialize the model with the given config and checkpoint on the
+        specific device.
+
+        Args:
+            cfg (ConfigType): Config contained the model information.
+            weights (str): Path to the checkpoint.
+            device (str, optional): Device to run inference. Defaults to 'cpu'.
+
+        Returns:
+            nn.Module: Model loaded with checkpoint.
+        """
+        model = MODELS.build(cfg.model)
+        load_checkpoint(model, weights, map_location='cpu')
+        model.cfg = cfg.model
+        model.to(device)
+        model.eval()
+        return model
+
+    def _init_collate(self, cfg: ConfigType) -> Callable:
+        """Initialize the ``collate_fn`` with the given config.
+
+        The returned ``collate_fn`` will be used to collate the batch data.
+        If will be used in :meth:`preprocess` like this
+
+        .. code-block:: python
+            def preprocess(self, inputs, batch_size, **kwargs):
+                ...
+                dataloader = map(self.collate_fn, dataloader)
+                yield from dataloader
+
+        Args:
+            cfg (ConfigType): Config which could contained the `collate_fn`
+                information. If `collate_fn` is not defined in config, it will
+                be :func:`pseudo_collate`.
+
+        Returns:
+            Callable: Collate function.
+        """
+        try:
+            collate_fn = COLLATE_FUNCTIONS.get(cfg.test_dataloader.collate_fn)
+        except AttributeError:
+            collate_fn = pseudo_collate
+        return collate_fn  # type: ignore
+
+    @abstractmethod
+    def _init_pipeline(self, cfg: ConfigType) -> Callable:
+        """Initialize the test pipeline.
+
+        Return a pipeline to handle varies of input data, such as str,
+        np.ndarray. It is an abstract method in BaseInferencer, and should be
+        implemented in subclasses.
+
+        The returned pipeline will be used to process a single data.
+        It will be used in :meth:`preprocess` like this:
+
+        .. code-block:: python
+            def preprocess(self, inputs, batch_size, **kwargs):
+                ...
+                dataset = map(self.pipeline, dataset)
+                ...
+        """
+        raise NotImplementedError('_init_pipeline is not implemented!')
```

Comment:
If using the `@abstractmethod` decorator, `raise NotImplementedError('_init_pipeline is not implemented!')` can be replaced as `pass`.

Reply:
If it is not necessary, maybe only providing the docstring is enough?